RoslynPluginGenerator/ZipExtensions.cs

```diff
@@ -58,4 +60,32 @@ public static void CreateFromDirectory(string sourceDirectoryName, string destin
             }
         }
     }
+
+    /// <summary>
+    /// Replaces the back slash to foward slash to attend the international convention of directories separation.
+    /// This will allow the zip file to be unzipped the correct way both linux and windows OS's.
+    /// </summary>
+    /// <seealso cref="System.Text.UTF8Encoding" />
+    class PathEncoder : UTF8Encoding
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathEncoder"/> class.
+        /// </summary>
+        public PathEncoder() : base(true)
+        {
+        }
+
+        /// <summary>
+        /// When overridden in a derived class, encodes all the characters in the specified string into a sequence of bytes.
+        /// </summary>
+        /// <param name="s">The string containing the characters to encode.</param>
+        /// <returns>
+        /// A byte array containing the results of encoding the specified set of characters.
+        /// </returns>
+        public override byte[] GetBytes(string s)
+        {
+            s = s.Replace("\\", "/");
```

Comment:
while this should work, I guess it would be better to use the `System.IO.Path.DirectorySeparatorChar` as replacement to make it fit to the operating system